### ISSUE TITLE
anyone tried --inspect with socketcluster

### ISSUE THREAD

**Issue by cometta** [SocketCluster/socketcluster#204]
I unable to run node --inspect with socketcluster .

$node --inspect ./server.babel.js
Debugger listening on port 9229.
Warning: This is an experimental feature and could change at any time.
To start debugging, open the following URL in Chrome:
    chrome-devtools://devtools/remote/serve_file/@521e5b7e2b7cc66b4006a8a54cb9c4e57494a5ef/inspector.html?experiments=true&v8only=true&ws=localhost:9229/node
Debugger listening on port 5859.
Warning: This is an experimental feature and could change at any time.
To start debugging, open the following URL in Chrome:
    chrome-devtools://devtools/remote/serve_file/@521e5b7e2b7cc66b4006a8a54cb9c4e57494a5ef/inspector.html?experiments=true&v8only=true&ws=localhost:5859/node
   [Busy] Launching SocketCluster
Unable to open devtools socket: address already in use
 1: node::Abort() [/usr/local/bin/node]
 2: node::inspector::AgentImpl::WorkerRunIO() [/usr/local/bin/node]
 3: uv__thread_start [/usr/local/bin/node]
 4: _pthread_body [/usr/lib/system/libsystem_pthread.dylib]
 5: _pthread_body [/usr/lib/system/libsystem_pthread.dylib]
 6: thread_start [/usr/lib/system/libsystem_pthread.dylib]

**Comment by mattkrick** [comment]
Heh, fun, I got a different callstack though:

```
   [Error] Error: scBroker server at socket path /var/folders/r_/sxm2m5wj5zl0k_z_6pczzydc0000gn/T/socketcluster/Action_cc2b3cba44/b0 exited
    at EventEmitter.<anonymous> (/Users/mk/Code/action2/node_modules/sc-broker-cluster/index.js:352:19)
    at emitTwo (events.js:106:13)
    at EventEmitter.emit (events.js:191:7)
    at EventEmitter.emit (/Users/mk/Code/action2/node_modules/sc-domain/index.js:12:31)
    at ChildProcess.<anonymous> (/Users/mk/Code/action2/node_modules/sc-broker/index.js:77:10)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at ChildProcess.EventEmitter.emit (/Users/mk/Code/action2/node_modules/sc-domain/index.js:12:31)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
```

@jondubois this is a gamechanger if we can get this to work, any ideas on where to start?
Must be at least node v6.3.0, I'm running v6.4.0

**Comment by jondubois** [comment]
@cometta @mattkrick Ok I implemented the fix in v5.0.9. Note that if you use the `--inspect` flag like `node --inspect server`, that will just debug the master process.

If you want to debug workers, you need to add an `--inspect-workers` flag AFTER the server argument like this: `node server --inspect-workers`.

You can also debug broker processes using `--inspect-workers` like this: `node server --inspect-brokers`.

I've updated the docs here http://socketcluster.io/#!/docs/debugging with info about the --inspect flag.

Note that there is currently a minor 'bug' in Node.js which prevents you from setting custom debug ports for the workers when using the `--inspect-workers=${startingPort}` flag. This is should not be a problem if you are happy to use the default ports.

I raised an issue on the Node.js repo: https://github.com/nodejs/node/issues/8201

**Comment by mattkrick** [comment]
Still getting something weird:

```
> NODE_ENV=development node ./src/server/server.babel.js --inspect-workers

   [Busy] Launching SocketCluster
   >> Broker PID: 52206
Debugger listening on port 5858.
Warning: This is an experimental feature and could change at any time.
To start debugging, open the following URL in Chrome:
    chrome-devtools://devtools/remote/serve_file/@62cd277117e6f8ec53e31b1be58290a6f7ab42ef/inspector.html?experiments=true&v8only=true&ws=localhost:5858/node
Debugger listening on port 5859.
Warning: This is an experimental feature and could change at any time.
To start debugging, open the following URL in Chrome:
    chrome-devtools://devtools/remote/serve_file/@62cd277117e6f8ec53e31b1be58290a6f7ab42ef/inspector.html?experiments=true&v8only=true&ws=localhost:5859/node
Creating a pool connected to localhost:28015
   >> Worker PID: 52208
   [Active] SocketCluster started
            Version: 5.0.9
            WebSocket engine: uws
            Port: 3000
            Master PID: 52205
            Worker count: 1
            Broker count: 1

Unable to open devtools socket: address already in use
Unable to open devtools socket: address already in use
Unable to open devtools socket: address already in use
1471749078806 - Origin: Worker (PID 52208)
   [Error] Error: channel closed
    at ChildProcess.target.send (internal/child_process.js:523:16)
    at Object.HappyThread.configure (/Users/mk/Code/action2/node_modules/happypack/lib/HappyThread.js:74:10)
    at /Users/mk/Code/action2/node_modules/happypack/lib/HappyThreadPool.js:39:18
    at /Users/mk/Code/action2/node_modules/happypack/node_modules/async/lib/async.js:713:13
    at async.forEachOf.async.eachOf (/Users/mk/Code/action2/node_modules/happypack/node_modules/async/lib/async.js:233:13)
    at _parallel (/Users/mk/Code/action2/node_modules/happypack/node_modules/async/lib/async.js:712:9)
    at Object.async.parallel (/Users/mk/Code/action2/node_modules/happypack/node_modules/async/lib/async.js:726:9)
    at Object.HappyThreadPool.configure (/Users/mk/Code/action2/node_modules/happypack/lib/HappyThreadPool.js:37:13)
    at /Users/mk/Code/action2/node_modules/happypack/lib/HappyPlugin.js:226:25
    at /Users/mk/Code/action2/node_modules/happypack/node_modules/async/lib/async.js:721:13
    at /Users/mk/Code/action2/node_modules/happypack/node_modules/async/lib/async.js:52:16
    at done (/Users/mk/Code/action2/node_modules/happypack/node_modules/async/lib/async.js:241:17)
    at /Users/mk/Code/action2/node_modules/happypack/node_modules/async/lib/async.js:44:16
    at /Users/mk/Code/action2/node_modules/happypack/node_modules/async/lib/async.js:718:17
    at /Users/mk/Code/action2/node_modules/happypack/node_modules/async/lib/async.js:167:37
    at /Users/mk/Code/action2/node_modules/happypack/lib/fnOnce.js:7:17
    at ChildProcess.<anonymous> (/Users/mk/Code/action2/node_modules/happypack/lib/HappyThread.js:22:11)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at ChildProcess.EventEmitter.emit (/Users/mk/Code/action2/node_modules/sc-domain/index.js:12:31)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
Unable to open devtools socket: address already in use
1471749078916 - Worker 0 exited - Exit code: 1
Debugger listening on port 5860.
Warning: This is an experimental feature and could change at any time.
To start debugging, open the following URL in Chrome:
    chrome-devtools://devtools/remote/serve_file/@62cd277117e6f8ec53e31b1be58290a6f7ab42ef/inspector.html?experiments=true&v8only=true&ws=localhost:5860/node
```

**Comment by jondubois** [comment]
@mattkrick You need to update dependencies too:
- sc-broker v2.2.2
- sc-broker-cluster v2.1.4

**Comment by cometta** [comment]
I tested this working fine now. May i know when we view the sourcemap , source file in chrome dev tool, it showed the babel transpiled code instead of raw es6 code. How to display es6 code in dev tool sourcemap?

**Comment by jondubois** [comment]
Note that recently there was a similar issue related to using the top-level `--inspect` flag to debug the master process. This has been fixed in socketcluster `v9.3.3`.

**Comment by marcj** [comment]
Note: This does not work with `ts-node`, only with `ts-node-dev`.

So

```
node_modules/.bin/ts-node src/main.ts --inspect-workers
```

would not work. Just use the ts-node-dev package.

```
node_modules/.bin/ts-node-dev src/main.ts --inspect-workers
```

and it prints something like 

```
Debugger listening on ws://127.0.0.1:5858/b2894045-ca5d-4b8a-ac3f-2594e980f655
For help, see: https://nodejs.org/en/docs/inspector
Debugger listening on ws://127.0.0.1:5859/a9193138-6e08-4c79-b3c1-980434a6a257
For help, see: https://nodejs.org/en/docs/inspector
```

which indicates it opened the debugging ports.
